### PR TITLE
Yast2 lan restarting -  	Feature 318787

### PR DESCRIPTION
**Pull request by dzedro**
I didn't add main.pm update, not sure how should it run as in my test or everywhere ?
http://10.100.98.90/tests/2921

### CODE REVIEW

**Comment by sysrich** [comment]
should be a good candidate for extratests IMHO

**Comment by dzedro** [comment]
yep EXTRATEST is good idea

**Comment by dzedro** [comment]
could look like this
http://10.100.98.90/tests/2924

File: tests/x11/yast2_lan_restart.pm [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+}
+
+sub check_network {
+    my $status = shift || 'no_restart';
+    wait_still_screen;
+    send_key 'alt-o';                           # OK
+    assert_screen 'yast2_closed_xterm_visible', 120;
+    script_run 'ip a';
+    # workaround bsc#992113
+    #assert_script_run 'dig suse.com|grep status';    # test if conection and DNS is working
```
Comment (okurz) [inline tests/x11/yast2_lan_restart.pm]: Don't add disabled code as nobody but you will understand why this is not active. Better leave it out and keep in your local git repository, either in `git stash` or a temporary "WIP"-commit

File: tests/x11/yast2_lan_restart.pm [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    if (check_screen 'yast2_lan_packages_need_to_be_installed', 5) {
+        send_key 'alt-i';              # Install
+    }
+    assert_screen 'yast2_closed_xterm_visible', 120;
+    script_run 'ip a';
+    script_run 'cat /dev/null > strace.log';    # clear strace.log
+}
+
+sub check_network {
+    my $status = shift || 'no_restart';
```
Comment (okurz) [inline tests/x11/yast2_lan_restart.pm]: can you write as

```perl
my ($status) = @_;
$status //= 'no_restart';
```
?

File: tests/x11/yast2_lan_restart.pm [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+    #assert_script_run 'dig suse.com|grep status';    # test if conection and DNS is working
+    if ("$status" eq 'restart') {
+        assert_script_run '[ -s strace.log ]';    # strace.log size is greater than zero (network restarted)
+    }
+    else {
+        script_run 'cat strace.log';                # print strace.log
+        assert_script_run '[ ! -s strace.log ]';    # strace.log size is not greater than zero (network not restarted)
+    }
+    script_run "cat strace.log > /dev/$serialdev";    # print strace.log
+    script_run 'cat /dev/null > strace.log';          # clear strace.log
```
Comment (okurz) [inline tests/x11/yast2_lan_restart.pm]: just `> strace.log' will do ;-)

File: tests/x11/yast2_lan_restart.pm [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+    }
+    elsif ("$device" eq 'bond') {
+        send_key 'alt-o';                             # Bond slaves
+        assert_screen 'yast2_lan_bond_slaves';
+        send_key 'tab';
+        send_key 'tab';
+        send_key 'tab';
+        send_key 'tab';
+        send_key 'tab';
+        send_key 'tab';
```
Comment (okurz) [inline tests/x11/yast2_lan_restart.pm]: wouldn't it be more robust to use `send_key_until_needlematch`?

File: tests/x11/yast2_lan_restart.pm (hunk 1 above)
Comment (dzedro) [inline tests/x11/yast2_lan_restart.pm]: that's the workaround that it's not testing connection, I just wanted to leave it there to don't forget

File: tests/x11/yast2_lan_restart.pm [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+}
+
+sub test_7 {
+    my $dev_name = shift;
+    diag '__________(7) Start yast2 lan -> Edit (a NIC) -> switch to Hardware tab, change nic name, [Next] -> [OK] -> device name is changed__________';
+    # workaround bsc#991694
+    record_soft_failure 'bsc#991694';
+    type_string "# workaround bsc#991694\n";
+    assert_script_run "wicked ifdown --delete eth0 sta0 dyn0";
+    # workaround bsc#991694
```
Comment (okurz) [inline tests/x11/yast2_lan_restart.pm]: I guess you are spelling this out a bit too often. IMHO the comments are not necessary when you already have the `record_soft_failure` *and* the `type_string` speaking for itself, right?

File: tests/x11/yast2_lan_restart.pm (hunk 4 above)
Comment (dzedro) [inline tests/x11/yast2_lan_restart.pm]: I expected this ;) but I'm not sure how robust is to match blinking cursor in text field

Comment (dzedro) [inline tests/x11/yast2_lan_restart.pm]: even worse not text field, but specific area which is not really highlighted when selected
http://10.100.98.90/tests/2928#step/yast2_lan_restart/168

File: tests/x11/yast2_lan_restart.pm (hunk 2 above)
Comment (dzedro) [inline tests/x11/yast2_lan_restart.pm]: Why not 

```
my $status = shift // 'no_restart';
```

Comment (coolo) [inline tests/x11/yast2_lan_restart.pm]: because accessing arguments with shift is just not modern perl

**Comment by dzedro** [comment]
Updated. http://10.100.98.90/tests/2932
I used selected tab, because selected area with devices is hard to see
http://10.100.98.90/tests/2930#step/yast2_lan_restart/182
```
        send_key_until_needlematch 'yast2_lan_bond_slave_tab_selected', 'tab';
        send_key 'tab';                               # select Bond Slaves and Order field
        send_key 'spc';                               # check network interface
```

**Pull request merged**